### ISSUE TITLE
Race condition in <Form /> with Firefox

### ISSUE THREAD

**Issue by m5r** [remix-run/remix#4780]
### What version of Remix are you using?

1.8.0

### Steps to Reproduce

- clone the minimal repro case: https://github.com/m5r/remix-form-firefox-redirect-bug
- create a [github oauth app](https://github.com/settings/applications/new) - make sure the field Authorization callback URL is set to `http://localhost:3000/sign-in/github/callback`
- generate a new client secret for this oauth app and copy the id & the secret into `the .env` file
- run the app
- sign in with github - this will take you through github's approval screen and then it redirects you back to the app
- sign out
- sign in again with github - this time, it doesnt go through github's approval screen and the app will never authenticate you

At this point, the remix app server will have logged something like this:
```
GET / 200 - - 7.426 ms
GET /sign-in?_data=routes%2Fsign-in 200 - - 4.052 ms
POST /sign-in/github 302 - - 5.879 ms
GET /sign-in/github/callback?code=f1944599380f04a97597&state=90d56e50-81cf-46f3-b392-ba71c359e9f2 - - - - ms
Error: queryRoute() call aborted
    at loadRouteData (/home/mokhtar/app/remix-form-firefox-redirect-bug/node_modules/@remix-run/router/router.ts:2278:13)
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
    at queryImpl (/home/mokhtar/app/remix-form-firefox-redirect-bug/node_modules/@remix-run/router/router.ts:2082:20)
    at Object.queryRoute (/home/mokhtar/app/remix-form-firefox-redirect-bug/node_modules/@remix-run/router/router.ts:2041:18)
    at handleResourceRequestRR (/home/mokhtar/app/remix-form-firefox-redirect-bug/node_modules/@remix-run/server-runtime/dist/server.js:265:20)
    at requestHandler (/home/mokhtar/app/remix-form-firefox-redirect-bug/node_modules/@remix-run/server-runtime/dist/server.js:46:18)
    at /home/mokhtar/app/remix-form-firefox-redirect-bug/node_modules/@remix-run/express/dist/server.js:39:22
GET /sign-in 200 - - 10.210 ms
```

### Expected Behavior

The form submission on `/sign-in/github` should wait for response from `/sign-in/github/callback`.  
It works as expected with Chrome but not with Firefox.

### Actual Behavior

With Firefox, the form submission on `/sign-in/github` receives a redirect to `/sign-in/github/callback` and it quickly interrupts it to reload the current page `/sign-in`.  
The behavior seems correct when JS is disabled though.

**Comment by jenseng** [comment]
Definitely seems like a weird browser-specific quirk, as your repro is using vanilla `<form>`, which should behave the same whether or not JS is enabled 😬

I haven't dug in, but I wonder if it's possible to create an even simpler repro, e.g. could this be reproduced without remix-auth, but instead with just actions within the app itself (e.g. a form post action with a delay and redirect)? Then you could create a remix PR that adds a conditionally failing test [here](https://github.com/remix-run/remix/blob/main/integration/form-test.ts) (e.g. see other examples of `test.fail` in that file).

**Comment by m5r** [comment]
That vanilla <form> might be an oversight from me as I was trying things out 😅  
I'll give it another shot tomorrow and see if I can make a Remix PR out of it

**Comment by m5r** [comment]
I managed to squeeze it in my evening and reproduced the bug in a test \o/  
I have found that redirecting within the app is fine, no matter how long a loader or an action takes. It's really the slow external connection that triggers it.

I didn't make a new test in `integration/form-test.ts` to avoid messing with the whole test suite. Instead I wrote my test in `integration/bug-report-test.ts`.  
It really does fail *only* in dev, in Firefox, with JS enabled :exploding_head:
